### PR TITLE
OOP and SQL refactoring

### PR DESCRIPTION
**Pull request by following5**
- cacheIcon.inc.php => cacheIcon.class.php & SQL refactoring
- geodb.inc.php => geodb.class.php & SQL refactoring
- locpics.inc.php => logpics.class.php & SQL refactoring
- removed obsolete require statements
- search.php SQL refactoring and security fix

<s>This PR is based on #292, which needs to be processed first. Will rebase if necessary.</s>

No further OOP refactoring planned, I mainly wanted to have the logpics thing in a class, before some [picture issues](http://redmine.opencaching.de/projects/oc-dev/issues?query_id=46) are done.

### CODE REVIEW

File: htdocs/adminhistory.php [hunk 1]
```diff
@@ -33,7 +32,7 @@
     $error = $translate->t('Cache not found', '', '', 0);
 } elseif ($cache_id > 0) {
     $showhistory = true;
-    $cache = new Cache($cache_id);
+    $cache = new cache($cache_id);
```
Comment (teiling88) [inline htdocs/adminhistory.php]: PSR-1 Issue: Class names MUST be declared in StudlyCaps.

**Comment by following5** [comment]
I will do another overhaul of the LogPics class, to get rid of those constanst ans switch.

**Comment by following5** [comment]
Did not work out as desired, but I did some other improvements. Template code should not be called here.

**Comment by teiling88** [comment]
pls rebase the branch after that it is ready to merge :-)

**Comment by following5** [comment]
I have rebased it and squashed two of the three commits.

**Pull request merged**